catalog/dags/providers/provider_api_scripts/auckland_museum.py

```diff
@@ -0,0 +1,201 @@
+)
+
+
+class AucklandMuseumDataIngester(ProviderDataIngester):
+    providers = {
+        "image": prov.AUCKLAND_MUSEUM_IMAGE_PROVIDER,
+    }
+    endpoint = "https://api.aucklandmuseum.com/search/collectionsonline/_search"
+    license_url = "https://creativecommons.org/licenses/by/4.0/"
+    total_amount_of_data = 10000
+    DEFAULT_LICENSE_INFO = get_license_info(license_url=license_url)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.delay = 4
+        self.batch_start = 0
+        self.batch_limit = 2000
+        self.headers = {"Content-Type": "application/json"}
+        date_from = datetime.strptime(self.date, "%Y-%m-%d")
+        self.date_from = date_from.isoformat()
+        self.date_to = (date_from + timedelta(days=1)).isoformat()
+        logger.info(f"Start timestamp: {self.date_from}, end timestamp: {self.date_to}")
+        self.data = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {"wildcard": {"copyright": {"value": "Auckland"}}},
+                        {"exists": {"field": "primaryRepresentation"}},
+                        {
+                            "range": {
+                                "lastModifiedOn": {
+                                    "from": self.date_from,
+                                    "to": self.date_to,
+                                }
+                            }
+                        },
+                    ]
+                }
+            }
+        }
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # Return default query params on the first request
+        # primaryRepresentation contain a image url for each data
+        # "+" is a query string syntax for must be present
+        # copyright:CC state Creative Commons Attribution 4.0
+        return {
+            "size": "2000",
+            "from": self.batch_start,
+        }
+
+    def get_batch_data(self, response_json):
+        # Takes the raw API response from calling `get` on the endpoint, and returns
+        # the list of records to process.
+        if response_json:
+            return response_json.get("hits", {}).get("hits")
+        return None
+
+    def get_should_continue(self, response_json):
+        # Do not continue if we have exceeded the total amount of data
+        self.batch_start += self.batch_limit
+        if self.batch_start >= self.total_amount_of_data:
+            logger.info(
+                "The final amount of data has been processed. Halting ingestion."
+            )
+            return False
+
+        return True
+
+    def get_media_type(self, record: dict):
+        return IMAGE
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        # check if _id is empty then foreign_landing_url and
+        # foreign_identifier doesn't exist
+
+        if not data.get("_id"):
```

Comment:
```suggestion
        if not (identifier := data.get("_id")):
```
We could simplify the code by saving the value of `_id` and reusing it below.